tests/smoke-tests/test-block-finalized.ts

```diff
@@ -0,0 +1,90 @@
+import "@moonbeam-network/api-augment";
+import { expect } from "chai";
+import { describeSmokeSuite } from "../util/setup-smoke-tests";
+
+const debug = require("debug")("smoke:block-finalized");
+const wssUrl = process.env.WSS_URL || null;
+const relayWssUrl = process.env.RELAY_WSS_URL || null;
+const ethRpcUrl = process.env.ETH_URL || null;
+
+describeSmokeSuite(
+  `Parachain blocks should be finalized..`,
+  { wssUrl, relayWssUrl, ethRpcUrl },
+  (context) => {
+    it("should have a recently finalized block", async function () {
+      const head = await context.polkadotApi.rpc.chain.getFinalizedHead();
+      const block = (await context.polkadotApi.rpc.chain.getBlock(head)).toHuman() as any;
+      const diff = Date.now() - getBlockTime(block);
+      debug(`Last finalized block was ${diff / 1000} seconds ago`);
+      expect(diff).to.be.lessThanOrEqual(10 * 60 * 1000); // 10 minutes in milliseconds
+    });
+
+    // When 0.9.29 is live we can enable this test
+    // TODO: update the CI to provide ETH_URL env var
+    it.skip("should have a recently finalized eth block", async function () {
+      const timestamp = (await context.ethers.getBlock("finalized")).timestamp;
+      const diff = Date.now() - timestamp * 1000;
+      debug(`Last finalized block was ${diff / 1000} seconds ago`);
+      expect(diff).to.be.lessThanOrEqual(10 * 60 * 1000);
+    });
+
+    it("should have only finalized blocks in the past two hours.", async function () {
+      this.timeout(120000);
+      this.slow();
+
+      const finalHash = await context.polkadotApi.rpc.chain.getFinalizedHead();
+      const block = (await context.polkadotApi.rpc.chain.getBlock(finalHash)).toHuman() as any;
+      const lastBlockNumber = Number(block.block.header.number.replace(/,/g, ""));
+      const lastBlockTime = getBlockTime(block);
+
+      // Target time here is set to be 2 hours, possible parameterize this in future
+      const firstBlockTime = lastBlockTime - 2 * 60 * 60 * 1000;
+      const firstBlockNumber = await fetchHistoricBlock(lastBlockNumber, firstBlockTime);
+      debug(`Checking if blocks #${firstBlockNumber} - #${lastBlockNumber} are finalized.`);
+
+      const promiseArray = range(firstBlockNumber, lastBlockNumber).map(async (num) => {
+        //@ts-ignore typescript doesn't like the custom RPC methods
```

Comment:
We should be able to provide those with our api-augment (`@moonbeam-network/api-augment`)

Reply:
Done, although it was only found in `@moonbeam-network/api-augment/moonbase`